go/vt/vtgate/planbuilder/dml.go

```diff
@@ -39,7 +39,17 @@ func buildUpdatePlan(upd *sqlparser.Update, vschema VSchema) (*engine.Route, err
 	route := &engine.Route{
 		Query: generateQuery(upd),
 	}
-	updateTable, _ := upd.Table.Expr.(sqlparser.TableName)
+	if len(upd.TableExprs) > 1 {
+		return nil, errors.New("unsupported: multi-table update")
+	}
+	aliased, ok := upd.TableExprs[0].(*sqlparser.AliasedTableExpr)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
+	updateTable, ok := aliased.Expr.(sqlparser.TableName)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
```

Comment:
Ok I added an initial implementation. Will add some tests soon. Let me know what you think